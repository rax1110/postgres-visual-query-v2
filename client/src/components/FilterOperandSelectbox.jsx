import { CustomInput, Row } from 'reactstrap';
import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import * as PropTypes from 'prop-types';
import { updateColumnOperand } from '../actions/queryActions';

const FilterOperandSelectbox = ({ column }) => {
  const dispatch = useDispatch();
  const [filterOperand, setFilterOperand] = useState('AND');

  useEffect(() => {
    dispatch(updateColumnOperand(filterOperand, column.id));
  }, [filterOperand, column.id, dispatch]);

  const handleChange = (e) => {
    setFilterOperand(e.target.value);
  };

  return (
    <Row form className="mb-2">
      <div className="ml-3">
        <CustomInput
          bsSize="sm"
          type="select"
          key={column.id}
          id="column_filter_operand"
          value={filterOperand}
          onChange={handleChange}
        >
          <option value="AND">AND</option>
          <option value="OR">OR</option>
          <option value="AND NOT">AND NOT</option>
          <option value="OR NOT">OR NOT</option>
        </CustomInput>
      </div>
    </Row>
  );
};

FilterOperandSelectbox.propTypes = {
  column: PropTypes.shape({ id: PropTypes.number }),
};

export default FilterOperandSelectbox;
