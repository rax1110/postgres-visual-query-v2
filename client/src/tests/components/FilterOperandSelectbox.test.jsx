import React from 'react';
import * as renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import FilterOperandSelectbox from '../../components/FilterOperandSelectbox';

describe('Component: FilterOperandSelectbox', () => {
  let store;
  let mockStore;
  let component;

  beforeEach(() => {
    const props = {
      column: {},
      updateColumnOperand: jest.fn(),
    };

    mockStore = configureStore([thunk]);
    store = mockStore(props);
    component = (
      <Provider store={store}>
        <FilterOperandSelectbox {...props} />
      </Provider>
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('FilterOperandSelectBox handleChange should set filter operand', () => {
    component = mount(component);

    component.find('CustomInput').simulate('change', {
      target: {
        value: 'OR',
      },
    });

    expect(component.find('CustomInput').prop('value')).toEqual('OR');
  });
});
