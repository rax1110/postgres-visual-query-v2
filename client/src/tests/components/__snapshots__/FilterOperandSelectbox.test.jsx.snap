// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FilterOperandSelectbox renders correctly 1`] = `
<div
  className="mb-2 form-row"
>
  <div
    className="ml-3"
  >
    <select
      className="custom-select custom-select-sm"
      id="column_filter_operand"
      onChange={[Function]}
      type="select"
      value="AND"
    >
      <option
        value="AND"
      >
        AND
      </option>
      <option
        value="OR"
      >
        OR
      </option>
      <option
        value="AND NOT"
      >
        AND NOT
      </option>
      <option
        value="OR NOT"
      >
        OR NOT
      </option>
    </select>
  </div>
</div>
`;
